tests/Unit/PointwiseFunctions/GeneralRelativity/Test_Christoffel.cpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "tests/Unit/PointwiseFunctions/GeneralRelativity/GrTestHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+void test_christoffel_first_kind(const DataVector& structure) {
+  const auto christoffel =
+      compute_christoffel_first_kind(make_spacetime_deriv_spacetime_metric(0.));
+
+  CHECK(christoffel.get(0, 0, 0) == approx(1.5));
+  CHECK(christoffel.get(0, 0, 1) == approx(2.));
+  CHECK(christoffel.get(0, 0, 2) == approx(2.5));
+  CHECK(christoffel.get(0, 0, 3) == approx(3.));
+  CHECK(christoffel.get(0, 1, 1) == approx(2.));
+  CHECK(christoffel.get(0, 1, 2) == approx(2.));
+  CHECK(christoffel.get(0, 1, 3) == approx(2.));
+  CHECK(christoffel.get(0, 2, 2) == approx(1.5));
+  CHECK(christoffel.get(0, 2, 3) == approx(1.));
+  CHECK(christoffel.get(0, 3, 3) == approx(0.));
+  CHECK(christoffel.get(1, 0, 0) == approx(4.));
+  CHECK(christoffel.get(1, 0, 1) == approx(6.));
+  CHECK(christoffel.get(1, 0, 2) == approx(8.));
+  CHECK(christoffel.get(1, 0, 3) == approx(10.));
+  CHECK(christoffel.get(1, 1, 1) == approx(8.));
+  CHECK(christoffel.get(1, 1, 2) == approx(10.));
+  CHECK(christoffel.get(1, 1, 3) == approx(12.));
+  CHECK(christoffel.get(1, 2, 2) == approx(12.));
+  CHECK(christoffel.get(1, 2, 3) == approx(14.));
+  CHECK(christoffel.get(1, 3, 3) == approx(16.));
+  CHECK(christoffel.get(2, 0, 0) == approx(6.5));
+  CHECK(christoffel.get(2, 0, 1) == approx(10.));
+  CHECK(christoffel.get(2, 0, 2) == approx(13.5));
+  CHECK(christoffel.get(2, 0, 3) == approx(17.));
+  CHECK(christoffel.get(2, 1, 1) == approx(14.));
+  CHECK(christoffel.get(2, 1, 2) == approx(18.));
+  CHECK(christoffel.get(2, 1, 3) == approx(22.));
+  CHECK(christoffel.get(2, 2, 2) == approx(22.5));
+  CHECK(christoffel.get(2, 2, 3) == approx(27.));
+  CHECK(christoffel.get(2, 3, 3) == approx(32.));
+  CHECK(christoffel.get(3, 0, 0) == approx(9.));
+  CHECK(christoffel.get(3, 0, 1) == approx(14.));
+  CHECK(christoffel.get(3, 0, 2) == approx(19.));
+  CHECK(christoffel.get(3, 0, 3) == approx(24.));
+  CHECK(christoffel.get(3, 1, 1) == approx(20.));
+  CHECK(christoffel.get(3, 1, 2) == approx(26.));
+  CHECK(christoffel.get(3, 1, 3) == approx(32.));
+  CHECK(christoffel.get(3, 2, 2) == approx(33.));
+  CHECK(christoffel.get(3, 2, 3) == approx(40.));
+  CHECK(christoffel.get(3, 3, 3) == approx(48.));
+
+  check_tensor_doubles_equals_tensor_datavectors(
+      compute_christoffel_first_kind(
+          make_spacetime_deriv_spacetime_metric(structure)),
+      christoffel);
+}
+
+SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.Christoffel.",
```

Comment:
test in each spatial dim and with both spatial and spacetime IndexType

Reply:
There have been previous bugs (some in SpEC, some in earlier versions of SpECTRE) where someone mistakenly hard-coded 3 as the spatial dimension (e.g. in a loop) for functions that should be valid in any spatial dimension.